mne_bids/tests/test_write.py

```diff
@@ -711,7 +706,8 @@ def test_edf(_bids_validate):
         read_raw_bids(bids_basename=bids_basename, bids_root=bids_root,
                       kind='eeg', extra_params=dict(foo='bar'))
 
-    bids_fname = bids_basename.replace('run-01', 'run-%s' % run2)
+    bids_fname = bids_basename.copy()
+    bids_fname.run = run2
```

Comment:
```suggestion
    bids_fname = bids_basename.copy().update(run=run2)
```

I would consistently use `update` everywhere

Reply:
Done.